### DISCUSSION TITLE
Open questions for next discussion

### DISCUSSION THREAD

**Discussion by AlCalzone** [zwave-js/zwave-js#1353] [General]
- [x] (https://github.com/zwave-js/node-zwave-js/discussions/1414) Configuration CC: getProperties(0) response invalid or no "next param #"
- [x] (https://github.com/zwave-js/node-zwave-js/discussions/1432) Improving startup times: What **needs** to be queried on startup?
- [x] (https://github.com/zwave-js/node-zwave-js/discussions/1416) What can we realistically do while waiting for a node response?
- [x] (https://github.com/zwave-js/node-zwave-js/discussions/1415) Do we need to expect every Get to fail? What if it does during the interview?
- [x] (https://github.com/zwave-js/node-zwave-js/discussions/1417) How to deal with missing/incorrect updates? E.g. get after set that returns the old value? How to reliably figure out if we need to do this at all?
- [x] (https://github.com/zwave-js/node-zwave-js/discussions/1415) What if the device's behavior contradicts what it (and the manual) claims to support? --> https://github.com/zwave-js/node-zwave-js/issues/1369 (ZW+ CC supported, no response to Get)
- [x] (https://github.com/zwave-js/node-zwave-js/discussions/1428) Dealing with sleeping nodes: Engaging contact, assumptions about awake status.
- [x] (https://github.com/zwave-js/node-zwave-js/discussions/1429) Dealing with `CAN`s: Retransmit (as per the specs) vs. duplicate responses

**Comment by LordMike** [comment]
>     * How to deal with missing/incorrect updates? E.g. get after set that returns the old value? How to reliably figure out if we need to do this at all?

Does this vary by device? .. else I kinda expect a REPORT to occur after a SET.. So maybe just assume the old value until a REPORT comes back. Otherwise associations wouldn't work - right.. ? :)

**Reply by AlCalzone** [reply]
This "discussion" serves as a question storage for me. You're free to ask questions, but please do that under a different category.

To answer your questions: A SET does not get a response. Some devices might report the updated value after a SET, many don't. Hence why many API methods have a GET after each SET currently. For some devices, this GET returns the new value, for some it returns the old or a transitional value.
